test/functional/users_controller_test.rb

```diff
@@ -309,6 +309,20 @@ def setup
     assert_equal I18n.t('users_controller.no_user_found'), flash[:error]
   end
 
+  test 'should get username in alphabetical order' do
+    get :list, params: { sort: 'username' }
+
+    assert_response :success
+    assert assigns(:users).each_cons(2).all?{|i,j| "j.node_revisions.username" >= "i.node_revisions.username" }
+  end
+
+  test 'should sort based on last activity in decreasing order' do
+    get :list, params: { sort: 'last_activity' }
+
+    assert_response :success
+    assert assigns(:users).each_cons(2).all?{|i,j| "i.updated_at" >= "j.updated_at" }
```

Comment:
Thanks for working on this @Manasa2850. With the test at line 316 and 323, should the signs `>=` be the inverse of each other? If alphabetical and ascending, then the i should be lower than the j? `j.node_revisions.username" <= "i.node_revisions.username"`

And for the last activity in descending order, it does make sense for the to be greater than j as you have written..

I haven't worked with that kind of comparison before, please let me know if what I am saying does not make sense.

Reply:
@RuthNjeri since we want the username to be sorted in ascending order, I've written `j.node_revisions.username" >= "i.node_revisions.username` where `j` refers to the second entry and `i` to the first one.
In line 323 it is `i.updated_at" >= "j.updated_at` which is the same as `j.updated_at" <= "i.updated_at` and is the reverse of what is there in line 316.

Reply:
Oh, I see, I think I missed the I and j difference, thanks for the clarification.

Reply:
@Manasa2850 Maybe here we can try with `i.node_revisions.maximum("timestamp")`

Reply:
Sure I'll try that out!

Reply:
@Tlazypanda it shows the same result `Expected false to be truthy` and the test passes on interchanging the values of `i` and `j`.

Reply:
hey @Manasa2850, I think removing  the quotes around the comparison  `assert assigns(:users).each_cons(2).all?{|i,j| i.updated_at >= j.updated_at }` should solve this, since its a time comparison. Thanks

Reply:
Thanks @cesswairimu! That worked :tada:

Reply:
great :tada: :tada: